src/brevitas/graph/calibrate.py

```diff
@@ -43,6 +43,20 @@ def finalize_collect_stats(module):
         module.counter = module.collect_stats_steps
 
 
+class do_calibration(object):
+    def __init__(self, model, enable=True):
+        self.model = model
+        self.current_state = model.training
+        self.disable_quant_inference = DisableQuantInference()
+        self.enable=enable
+
+    def __enter__(self):
```

Comment:
This way when enabled=False it stills puts the model in training mode. It should be more something like: 
__enter__:
if enabled:
    self.disable_quant_inference.apply(self.model, is_training=True, quantization_enabled=False)

__exit__:
    self.disable_quant_inference.apply(self.model, is_training=current_state, quantization_enabled=True)